### PR TITLE
[SDK-681] Change to latest setup

### PR DESCRIPTION
**Pull request by bitPogo**
### :pushpin: References
* **Issue:** [SDK-681](https://gesundheitscloud.atlassian.net/browse/SDK-681)

### :tophat: What is the goal?

Update project config to latest iteration of a SDK setup

### :unicorn: How is it being implemented?

Updated docs, github config, project config and gradlePlugin setup

### :thinking: DOD Checklist

- [x] My code follows the code style and naming conventions of this project.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated the documentation.
- [ ] I have updated the changelog accordingly.

### CODE REVIEW

File: buildSrc/src/main/kotlin/care/data4life/gradle/fhir/script/download-scripts.gradle.kts [hunk 1]
```diff
@@ -58,7 +58,7 @@ private val repository = "https://raw.githubusercontent.com/d4l-data4life/hc-gra
 private val branch = "main"
 val baseLink = "$repository/$branch"
 val scriptPathSource = "buildSrc/src/main/kotlin/care/data4life/sdk/script"
-val scriptPathTarget = "buildSrc/src/main/kotlin/care/data4life/hl7/fhir"
+val scriptPathTarget = "buildSrc/src/main/kotlin/care/data4life/gradle/datadonation/script"
```
Comment (wmontwe) [inline buildSrc/src/main/kotlin/care/data4life/gradle/fhir/script/download-scripts.gradle.kts:61]: should not be `datadonation` -> `fhir`

File: gradlePlugin/fhir-dependency/src/main/kotlin/care/data4life/gradle/fhir/dependency/Dependency.kt [hunk 2]
```diff
@@ -54,39 +54,39 @@ object Dependency {
             }
 
             object Coroutines {
-                const val common = "org.jetbrains.kotlinx:kotlinx-coroutines-core:${Version.multiplatform.kotlin.coroutines}"
-                const val android = "org.jetbrains.kotlinx:kotlinx-coroutines-android:${Version.multiplatform.kotlin.coroutines}"
-                const val test = "org.jetbrains.kotlinx:kotlinx-coroutines-test:${Version.multiplatform.kotlin.coroutines}"
+                const val common = "org.jetbrains.kotlinx:kotlinx-coroutines-core:${Version.Multiplatform.Kotlin.coroutines}"
```
Comment (wmontwe) [inline gradlePlugin/fhir-dependency/src/main/kotlin/care/data4life/gradle/fhir/dependency/Dependency.kt:57]: converted from small to big letters

File: DEVELOPING.adoc [hunk 3]
```diff
@@ -26,8 +26,6 @@ toc::[]
 * link:https://developer.android.com/studio#downloads[Android Studio 4.2.1]
 * link:https://adoptopenjdk.net/?variant=openjdk8&jvmVariant=hotspot[Java 8] | link:https://developer.android.com/studio/write/java8-support[Limitations of Java language features and APIs] | https://jakewharton.com/d8-library-desugaring/[Desugaring]
 * link:https://kotlinlang.org/[Kotlin 1.4.32]
-* link:https://www.python.org/downloads/[Python 3.7.8]
```
Comment (wmontwe) [inline DEVELOPING.adoc:29]: This is necessary for the project and also the following descriptions

File: DEVELOPING.adoc [hunk 4]
```diff
@@ -64,7 +62,7 @@ Just install the mentioned dependencies:
 
 [source,bash]
 ----
-sdk install java 8.0.292.hs-adpt
+sdk install java  8.0.292.hs-adpt
```
Comment (wmontwe) [inline DEVELOPING.adoc:65]: spacing too much

File: DEVELOPING.adoc [hunk 5]
```diff
@@ -76,37 +74,10 @@ java -version
 kotlin -version
 ----
 
-=== Install Python
```
Comment (wmontwe) [inline DEVELOPING.adoc:79]: need to be kept

File: DEVELOPING.adoc [hunk 6]
```diff
@@ -76,37 +74,10 @@ java -version
...
-[source,bash]
-----
-pyenv install
-----
-
 === Android Studio
 
 For better interop with iOS you may install the link:https://plugins.jetbrains.com/plugin/14936-kotlin-multiplatform-mobile[Kotlin Multiplaform Mobile (KMM) plugin for Android Studio]. It let's you run, test and debug shared code on Android and iOS without switching the IDEs.
 
-=== First time
```
Comment (wmontwe) [inline DEVELOPING.adoc:101]: need to be kept

File: DEVELOPING.adoc [hunk 7]
```diff
@@ -150,45 +121,4 @@ Releases are automatically created when a tag in the form of `v{major}.{minor}.{
 
 == Develop
 
-The source is divided into supporting code for JSON parsing and establishing the FHIR primitive type system (String, Dates, Integer, etc.) for link:http://hl7.org/fhir/STU3/datatypes.html[FHIR 3] and link:http://hl7.org/fhir/R4/datatypes.html[FHIR 4]
```
Comment (wmontwe) [inline DEVELOPING.adoc:153]: need to be kept

**Review by wmontwe — CHANGES_REQUESTED** [review]
Some little findings

File: buildSrc/src/main/kotlin/care/data4life/gradle/fhir/script/download-scripts.gradle.kts (hunk 1 above)
Comment (bitPogo) [reply buildSrc/src/main/kotlin/care/data4life/gradle/fhir/script/download-scripts.gradle.kts:61]: Glad you spotted it!!!!

Comment (bitPogo) [reply buildSrc/src/main/kotlin/care/data4life/gradle/fhir/script/download-scripts.gradle.kts]: In the line before - It must be also gradle from now on, right?

**Review by wmontwe — APPROVED** [review]
👍

File: buildSrc/src/main/kotlin/care/data4life/gradle/fhir/script/download-scripts.gradle.kts (hunk 1 above)
Comment (wmontwe) [reply buildSrc/src/main/kotlin/care/data4life/gradle/fhir/script/download-scripts.gradle.kts]: 👍

**Pull request merged**